iati/tests/test_codelists.py

```diff
@@ -182,71 +182,53 @@ def test_code_enumeration_element(self):
-        """Return a comparison function that checks whether things are equal."""
-        return request.param
-
-    @pytest.fixture(params=[
-        lambda x, y: x != y,
-        lambda x, y: y != x,
-        lambda x, y: hash(x) != hash(y)
-    ])
-    def cmp_func_different(self, request):
-        """Return a comparison function that checks whether things are different."""
-        return request.param
-
     @pytest.mark.parametrize('codelist', iati.default.codelists('2.02').values())
-    def test_codelist_same_object_equal(self, codelist, cmp_func_equal):
+    def test_codelist_same_object_equal(self, codelist, cmp_func_equal_val_and_hash):
         """Check that a Codelist is deemed to be equal with itself."""
-        assert cmp_func_equal(codelist, codelist)
+        assert cmp_func_equal_val_and_hash(codelist, codelist)
 
     @pytest.mark.parametrize('codelist', iati.default.codelists('2.02').values())
-    def test_codelist_same_diff_object_equal(self, codelist, cmp_func_equal):
+    def test_codelist_same_diff_object_equal(self, codelist, cmp_func_equal_val_and_hash):
         """Check that two instances of the same Codelist are deemed to be equal."""
         codelist_copy = copy.deepcopy(codelist)
 
-        assert cmp_func_equal(codelist, codelist_copy)
+        assert cmp_func_equal_val_and_hash(codelist, codelist_copy)
 
     @pytest.mark.parametrize('codelist', iati.default.codelists('2.02').values())
-    def test_codelist_diff_name_not_equal(self, codelist, cmp_func_different):
+    def test_codelist_diff_name_not_equal(self, codelist, cmp_func_different_val_and_hash):
         """Check that two different Codelists are not deemed to be equal.
 
         The two Codelists have different names, but are otherwise identical.
         """
         codelist_copy = copy.deepcopy(codelist)
         codelist_copy.name = codelist.name + 'with a difference'
 
-        assert cmp_func_different(codelist, codelist_copy)
+        assert cmp_func_different_val_and_hash(codelist, codelist_copy)
 
     @pytest.mark.parametrize('codelist', iati.default.codelists('2.02').values())
-    def test_codelist_diff_completeness_not_equal(self, codelist, cmp_func_different):
+    def test_codelist_diff_completeness_not_equal(self, codelist, cmp_func_different_val_and_hash):
         """Check that two different Codelists are not deemed to be equal.
 
         The two Codelists have different completeness, but are otherwise identical.
         """
         codelist_copy = copy.deepcopy(codelist)
         codelist_copy.complete = not codelist.complete
 
-        assert cmp_func_different(codelist, codelist_copy)
+        assert cmp_func_different_val_and_hash(codelist, codelist_copy)
 
     @pytest.mark.parametrize('codelist', iati.default.codelists('2.02').values())
-    def test_codelist_diff_num_codes_not_equal(self, codelist, cmp_func_different):
+    def test_codelist_diff_num_codes_not_equal(self, codelist, cmp_func_different_val_and_hash):
```

Comment:
Is there a side issue here to test for non-2.02 verisons?

Reply:
And elsewhere in this file...

Reply:
There should be!

Reply:
#257 created